pkg/maps/ctmap/ctmap.go

```diff
@@ -360,24 +361,38 @@ func (f *GCFilter) doFiltering(dstIP net.IP, dstPort uint16, nextHdr, flags uint
 		action = deleteEntry
 	}
 
-	// used by FlushCTEntriesOf
-	// Delete all entries of the given dstIP that are not filtered
-	if f.Type&GCFilterByIDsToKeep != 0 &&
-		dstIP.Equal(f.IP) &&
-		// only delete TUPLE_F_IN connections since the clean up is made
-		// for ingress policies
-		flags&TUPLE_F_IN != 0 {
+	// If the filter doesn't contain an endpoint ID & IP, no entries will get matched below.
+	if f.EndpointID == 0 || f.EndpointIP == nil {
+		return
+	}
+
+	// Determine whether the entry matches the endpoint IP,
+	// and the direction of the entry (ingress or egress).
+	var ingress bool
+	if flags == TUPLE_F_IN && dstIP.Equal(f.EndpointIP) {
+		ingress = true
+	} else if flags == TUPLE_F_OUT && srcIP.Equal(f.EndpointIP) {
```

Comment:
This isn't handling the `flags &= TUPLE_F_RELATED` case.

Typically we have three conntrack entries per connection:
* A -> B, proto, dir
* B -> A, proto, dir
* B -> A, ICMP (related)